lib/config/src/app/job.rs

```diff
@@ -0,0 +1,262 @@
+use std::{borrow::Cow, collections::HashMap, fmt::Display, str::FromStr};
+
+use serde::{de::Error, Deserialize, Serialize};
+
+use crate::package::PackageSource;
+
+use super::{pretty_duration::PrettyDuration, AppConfigCapabilityMemoryV1, AppVolume, HttpRequest};
+
+/// Job configuration.
+#[derive(
+    serde::Serialize, serde::Deserialize, schemars::JsonSchema, Clone, Debug, PartialEq, Eq,
+)]
+pub struct Job {
+    name: String,
+    trigger: JobTrigger,
+    #[serde(flatten)]
+    action: JobAction,
+}
+
+#[derive(
+    serde::Serialize, serde::Deserialize, schemars::JsonSchema, Clone, Debug, PartialEq, Eq,
+)]
+#[serde(rename_all = "lowercase")]
+pub enum JobAction {
+    Fetch(HttpRequest),
+    Execute(ExecutableJob),
+}
+
+#[derive(
+    serde::Serialize, serde::Deserialize, schemars::JsonSchema, Clone, Debug, PartialEq, Eq,
+)]
+pub enum CronType {
+    Daily,
+    Hourly,
+    Weekly,
+    CronExpression(String),
+}
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum JobTrigger {
+    PreDeployment,
+    PostDeployment,
+    Cron(CronType),
+}
+
+#[derive(
+    serde::Serialize, serde::Deserialize, schemars::JsonSchema, Clone, Debug, PartialEq, Eq,
+)]
+pub struct ExecutableJob {
+    /// The package that contains the command to run. Defaults to the app config's package.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    package: Option<PackageSource>,
+
+    /// The command to run. Defaults to the package's entrypoint.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    command: Option<String>,
+
+    /// CLI arguments passed to the runner.
+    /// Only applicable for runners that accept CLI arguments.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    cli_args: Option<Vec<String>>,
+
+    /// Environment variables.
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub env: Option<HashMap<String, String>>,
+
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub capabilities: Option<ExecutableJobCompatibilityMapV1>,
+
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub volumes: Option<Vec<AppVolume>>,
+
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub timeout: Option<PrettyDuration>,
```

Comment:
These settings (timeout,drift,retries) should be on the cron, not on this action.
They are also relevant for fetch jobs.

Reply:
Nice catch!